### PR TITLE
feat: test ERC721

### PR DESCRIPTION
**Pull request by davidprepo**

### CODE REVIEW

**Review by ramenforbreakfast — APPROVED** [review]
LGTM

File: apps/smart-contracts/core/contracts/test/TestERC721.sol [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+contract TestERC721 is ERC721 {
+  uint256 tokenId = 1;
+
+  constructor(string memory name_, string memory symbol_)
+    ERC721(name_, symbol_)
+  {}
+
+  function mint(address _recipient) external {
+    _mint(_recipient, tokenId);
+    tokenId += 1;
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/contracts/test/TestERC721.sol:15]: nit:
```suggestion
    tokenId++;
```

File: apps/smart-contracts/core/contracts/test/TestERC721.sol [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity =0.8.7;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+
+contract TestERC721 is ERC721 {
+  uint256 tokenId = 1;
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/contracts/test/TestERC721.sol:7]: can just start at 0
```suggestion
  uint256 tokenId;
```

**Review by XavierEkkel — APPROVED** [review]
LGTM after comments

**Pull request merged**